### PR TITLE
Fix catchup with startingFrom-FactId

### PR DESCRIPTION
**Pull request by uweschaefer**
see #307

### CODE REVIEW

**Comment by StephanPraetsch** [comment]
I do see an improved thread-safe behavior and a test that describes the bug.
But I don't get it what caused that bug and why it's fixed now. Could you provide some information, please?

**Comment by uweschaefer** [comment]
subscribe(to,observer) called doCatchup twice, once without a lock, then a second time with the lock. this idea was to not lock for too long unnecessarily.

The problem was, that AfterPredicate is used on catchup, which uses a flipswitch.
When reused, that switch is open from the very beginning, so that on the second catchup, you'll get anything, not just the Facts after the "from" one.

This can be further improved by not reusing the predicate of course, but just not catching up twice did the trick and was the cheapest patch to do.

File: factcast-store-inmem/src/main/java/org/factcast/store/inmem/InMemFactStore.java [hunk 1]
```diff
@@ -179,11 +181,6 @@ public synchronized Subscription subscribe(SubscriptionRequestTO request,
         executorService.submit(() -> {
             // catchup
             AtomicLong ser = new AtomicLong(-1);
-            if (!request.ephemeral()) {
```
Comment (uweschaefer) [inline factcast-store-inmem/src/main/java/org/factcast/store/inmem/InMemFactStore.java]: deleting this fixed the bug. @StephanPraetsch

**Comment by uweschaefer** [comment]
i think i cannot wait for review any longer - otherwise the branch becomes too old.

**Pull request merged**